cmd/disk-cache.go

```diff
@@ -186,6 +186,9 @@ func (c cacheObjects) getMetadata(objInfo ObjectInfo) map[string]string {
 func (c cacheObjects) GetObjectNInfo(ctx context.Context, bucket, object string, rs *HTTPRangeSpec, h http.Header, lockType LockType, opts ObjectOptions) (gr *GetObjectReader, err error) {
 
 	objInfo, bkErr := c.GetObjectInfoFn(ctx, bucket, object, opts)
+	if bkErr != nil {
+		return nil, bkErr
+	}
```

Comment:
@leitzler, good catch. It would be better to move the GetObjectInfoFn after getting the cache location in lines 195-198 and move the objInfo.IsCacheable() down to line 209. If the error from GetObjectInfo is due to object being deleted from the backend, we would need to clean up stale cache entries right away. Pls check the GetObject implementation which is doing it correctly.

Reply:
Good idea! I moved it down and also noticed that GetObject doesn't check `objInfo.IsCachable()`, is that intended?

Check out my latest commit to see if that is in line of your idea!

Reply:
@leitzler, GetObject has been replaced by GetObjetcNInfo.But since it is still in the codebase, pls add a check for objInfo.IsCachable() in GetObject